crates/uv-python/src/managed.rs

```diff
@@ -681,19 +681,26 @@ impl ManagedPythonInstallation {
         if (self.key.major, self.key.minor) != (other.key.major, other.key.minor) {
             return false;
         }
-        // Require a newer, or equal patch version (for pre-release upgrades)
-        if self.key.patch < other.key.patch {
-            return false;
-        }
-        if let Some(other_pre) = other.key.prerelease {
-            if let Some(self_pre) = self.key.prerelease {
+        // If the patch versions are the same, we're handling a pre-release upgrade
+        if self.key.patch == other.key.patch {
+            // Require a newer pre-release, if present on both
+            if let Some(other_pre) = other.key.prerelease
+                && let Some(self_pre) = self.key.prerelease
+            {
                 return self_pre > other_pre;
             }
-            // Do not upgrade from non-prerelease to prerelease
+            // Allow upgrade from pre-release to stable
+            if self.key.prerelease.is_none() && other.key.prerelease.is_some() {
+                return true;
+            }
+            // Do not upgrade from pre-release to stable, or for matching versions
             return false;
```

Comment:
Trying to decide if I like this better:
```suggestion
            return match (self.key.prerelease, other.key.prerelease) {
                // Require a newer pre-release, if present on both
                (Some(self_pre), Some(other_pre)) => self.pre > other.pre,
                // Allow upgrade from pre-release to stable
                (None, Some(_)) => true,
                // Do not upgrade from pre-release to stable, or for matching versions
                (_, None) = > false,
            }
```

Reply:
I don't mind that